src/Controller/BlogController.php

```diff
@@ -47,15 +48,14 @@ class BlogController extends AbstractController
      * Content-Type header for the response.
      * See https://symfony.com/doc/current/quick_tour/the_controller.html#using-formats
      */
-    public function index(int $page, string $_format): Response
+    public function index(int $page, string $_format, PostRepository $posts): Response
     {
-        $em = $this->getDoctrine()->getManager();
-        $posts = $em->getRepository(Post::class)->findLatest($page);
+        $latestPosts = $posts->findLatest($page);
 
         // Every template name also has two extensions that specify the format and
         // engine for that template.
         // See https://symfony.com/doc/current/templating.html#template-suffix
-        return $this->render('blog/index.'.$_format.'.twig', ['posts' => $posts]);
+        return $this->render('blog/index.'.$_format.'.twig', ['posts' => $latestPosts]);
```

Comment:
What about just `['posts' => $posts->findLatest($page)]`?

Reply:
Not sure ... the line would be a bit long.